dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java

```diff
@@ -0,0 +1,388 @@
+import static com.dotcms.util.CollectionsUtils.imap;
+import com.dotcms.contenttype.business.FieldAPI;
+import com.dotcms.contenttype.exception.NotFoundInDbException;
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.contenttype.transform.field.JsonFieldTransformer;
+import com.dotcms.repackage.com.google.common.annotations.VisibleForTesting;
+import com.dotcms.repackage.javax.ws.rs.Consumes;
+import com.dotcms.repackage.javax.ws.rs.DELETE;
+import com.dotcms.repackage.javax.ws.rs.GET;
+import com.dotcms.repackage.javax.ws.rs.POST;
+import com.dotcms.repackage.javax.ws.rs.PUT;
+import com.dotcms.repackage.javax.ws.rs.Path;
+import com.dotcms.repackage.javax.ws.rs.PathParam;
+import com.dotcms.repackage.javax.ws.rs.Produces;
+import com.dotcms.repackage.javax.ws.rs.core.Context;
+import com.dotcms.repackage.javax.ws.rs.core.MediaType;
+import com.dotcms.repackage.javax.ws.rs.core.Response;
+import com.dotcms.repackage.org.glassfish.jersey.server.JSONP;
+import com.dotcms.rest.InitDataObject;
+import com.dotcms.rest.ResponseEntityView;
+import com.dotcms.rest.WebResource;
+import com.dotcms.rest.annotation.NoCache;
+import com.dotcms.rest.api.v1.authentication.ResponseUtil;
+import com.dotcms.rest.exception.ForbiddenException;
+import com.dotcms.rest.exception.mapper.ExceptionMapperUtil;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.model.User;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+
+
+@Path("/v2/contenttype/{typeId}/fields")
+public class FieldResource implements Serializable {
+    private final WebResource webResource;
+    private final FieldAPI fieldAPI;
+
+    public FieldResource() {
+        this(new WebResource(), APILocator.getContentTypeFieldAPI());
+    }
+
+    @VisibleForTesting
+    public FieldResource(final WebResource webresource, final FieldAPI fieldAPI) {
+        this.fieldAPI = fieldAPI;
+        this.webResource = webresource;
+    }
+
+    private static final long serialVersionUID = 1L;
+
+
+    @PUT
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces({ MediaType.APPLICATION_JSON, "application/javascript" })
+    public Response updateFields(@PathParam("typeId") final String typeId, final String fieldsJson,
+            @Context final HttpServletRequest req) throws DotDataException, DotSecurityException {
+
+        final InitDataObject initData = this.webResource.init(null, false, req, false, null);
+        final User user = initData.getUser();
+
+        Response response = null;
+
+        try {
+            final List<Field> fields = new JsonFieldTransformer(fieldsJson).asList();
+
+            for (final Field field : fields) {
```

Comment:
I think fieldAPI is transactional, I am wondering if you want to handle each as a trx or follow an all or nothing approach. If you do, probably have to wrap the for into a method and annotated as WrapInTransaction